service/r2/service.go

```diff
@@ -553,3 +565,37 @@ type ruleSetJSON struct {
 	MappingRules  []mappingRuleJSON `json:"mappingRules"`
 	RollupRules   []rollupRuleJSON  `json:"rollupRules"`
 }
+
+// Creates a new RuleSetSnapshot from a rulesetJSON. If the ruleSetJSON has no IDs for any of its
+// mapping rules or rollup rules, it generates missing IDs and sets as a string UUID string so they
+// can be stored in a mapping (id -> rule).
```

Comment:
If a rule is being added, not updated is that when there would not be an ID for a rule? I might update the comment to explain what the scenario is where this happens.

Reply:
Yeah that's exactly it - I'll add a comment for this scenario.

Reply:
nit: Change the comment to `ruleSetSnapshot creates a ...`.

You might also make this behavior controlled by an enum passed in though (e.g., genIDType), e.g., if this is an update request, we probably shouldn't generate the missing IDs since that should be a validation error.